src/herder/Upgrades.cpp

```diff
@@ -238,7 +238,7 @@ Upgrades::removeUpgrades(std::vector<UpgradeType>::const_iterator beginUpdates,
 bool
 Upgrades::isValid(uint64_t closeTime, UpgradeType const& upgrade,
                   LedgerUpgradeType& upgradeType, bool nomination,
-                  Config const& cfg) const
+                  Config const& cfg, LedgerHeader const& header) const
```

Comment:
what about passing just ledgerVersion here? we are not using full LedgerHeader struct here

Reply:
Conceptually it should take a `ledgerHeader` as a parameter as upgrades perform arbitrary changes based on the ledger header (I can see having conditions only enforced after a certain version number, or re-introduce some of the max-rate change to fee/base reserve logic we removed as part of the transition but that can be re-added now that we formalized upgrades), I guess the bug in the interface is that it takes "closeTime" (and we'll be chasing all fields from the ledger header over time), let me fix this.

Reply:
This also brings back the symmetry between `createUpgradesFor` and `isValid` (where the outside world should not have to know about the internals of what upgrades do)